easybuild/framework/easyblock.py

```diff
@@ -897,8 +895,10 @@ def obtain_file(self, filename, extension=False, urls=None, download_filename=No
                     source_urls = []
                 source_urls.extend(self.cfg['source_urls'])
 
-                # add https://sources.easybuild.io as fallback source URL
-                source_urls.append(EASYBUILD_SOURCES_URL + '/' + os.path.join(name_letter, location))
+                # Add additional URLs as configured.
+                for url in build_option("extra_source_urls") or []:
```

Comment:
Looks like f4ec1825c740410f667342caa3403a4e37657c33 is not required after you fixed the issue with 1f5542e586ad3661c56d7394786627f6661a5bc5, i.e. "extra_source_urls" can never be None hence no need for the `or []`

I'd suggest to either revert that commit or rebase skipping it. Otherwise people might get confused later how it could be anything but a (possibly) empty list. But it can't even be an **empty** list, can it?

Or:
```suggestion
                for url in build_option("extra_source_urls"):
```

Reply:
My understanding is that it cannot be an empty list now it has a properly listed default value in `BUILD_OPTIONS_CMDLINE`, indeed. I'll remove it.

You mentioned rebasing. Would you prefer I squash everything to one commit here? Or do you want to keep the review history?

Reply:
That is correct, yes

I'm not allowed to merge anyway so I can't really decide it. But my person preference for this is to squash. Most commits are just stylistic anyway, not some redesign or something. Only thing would be https://github.com/easybuilders/easybuild-framework/pull/4079/commits/1f5542e586ad3661c56d7394786627f6661a5bc5 to keep a mention why it is put there but even that is "obvious" enough. Hence a single commit is fine for me.

But I know that others like to keep all history or just don't want to do the extra work of squashing.